### PR TITLE
[WIPTEST] Cluster should require provider obj

### PR DESCRIPTION
**Pull request by izapolsk**
Purpose
=================
1. Cluster requires provider now;
2. get_all_clusters moved to Provider object;
3. all usage places are fixed;

{{pytest: cfme/tests/infrastructure/test_delete_infra_object.py cfme/tests/infrastructure/test_timelines.py cfme/tests/infrastructure/test_cluster_analysis.py}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit d48db1a2162991c88188ef03d1c4d1a688db385b:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/infrastructure/cluster.py [hunk 1]
```diff
@@ -4,15 +4,15 @@
 :var page: A :py:class:`cfme.web_ui.Region` object describing common elements on the
            Cluster pages.
 """
+from functools import partial
```
Comment (psav) [inline cfme/infrastructure/cluster.py]: Should be a line break between the built-ins, thirdparty, and in-repo sets of imports

File: cfme/infrastructure/provider/__init__.py [hunk 2]
```diff
@@ -56,11 +56,10 @@
         ('ipaddress_text', Input("ipaddress"), {"removed_since": "5.4.0.0.15"}),
         ('api_port', Input("port")),
         ('sec_protocol', {version.LOWEST: Select("select#security_protocol"),
-            '5.5': AngularSelect("security_protocol", exact=True)}),
+                          '5.5': AngularSelect("security_protocol", exact=True)}),
```
Comment (psav) [inline cfme/infrastructure/provider/__init__.py]: This is fine as is, we prefer to always try to indent multiples of 4

File: cfme/infrastructure/provider/__init__.py [hunk 3]
```diff
@@ -102,18 +100,18 @@
...
-                  'Discover Infrastructure Providers'),
-              'infrastructure_provider': [lambda ctx: sel.click(Quadicon(ctx['provider'].name,
-                                                                         'infra_prov')),
-                                          {'infrastructure_provider_edit':
-                                           lambda _: cfg_btn('Edit this Infrastructure Provider'),
-                                           'infrastructure_provider_policy_assignment':
-                                           lambda _: pol_btn('Manage Policies'),
-                                           'infrastructure_provider_timelines':
-                                           lambda _: mon_btn('Timelines')}]})
+               {'infrastructure_provider_new': lambda _: cfg_btn(
```
Comment (psav) [inline cfme/infrastructure/provider/__init__.py]: I'll merge this in, but we don't care as it's going soon :)

File: cfme/infrastructure/provider/__init__.py [hunk 4]
```diff
@@ -285,6 +283,15 @@ def hosts(self):
             result.append(Host(name=host["name"], credentials=cred))
         return result
 
+    def get_clusters(self):
+        """returns the list of clusters belonging to the provider"""
+        web_clusters = []
+        sel.force_navigate('infrastructure_clusters')
```
Comment (psav) [inline cfme/infrastructure/provider/__init__.py]: Doesn't this return _all_ clusters? ie, it doesn't seem to filter at all to show _just_ the providers clusters? Did I miss something?

File: cfme/tests/infrastructure/test_cluster_analysis.py [hunk 5]
```diff
@@ -35,7 +35,7 @@ def test_run_cluster_analysis(request, setup_provider, provider, remove_test, so
         test_flag: cluster_analysis
     """
     cluster_name = remove_test['cluster']
-    test_cluster = cluster.Cluster(name=cluster_name)
+    test_cluster = cluster.Cluster(name=cluster_name, provider=provider)
```
Comment (psav) [inline cfme/tests/infrastructure/test_cluster_analysis.py]: Great!

File: cfme/infrastructure/cluster.py (hunk 1 above)
Comment (izapolsk) [inline cfme/infrastructure/cluster.py]: agree. I also use such approach but I also try to follow code conventions existing in this project.
fixing

File: cfme/infrastructure/provider/__init__.py (hunk 2 above)
Comment (izapolsk) [inline cfme/infrastructure/provider/__init__.py]: I just follow Pycharm's Pep8 recommendations.

File: cfme/infrastructure/provider/__init__.py (hunk 3 above)
Comment (izapolsk) [inline cfme/infrastructure/provider/__init__.py]: I see but this is just Pycharm's auto-indention.

File: cfme/infrastructure/provider/__init__.py (hunk 4 above)
Comment (izapolsk) [inline cfme/infrastructure/provider/__init__.py]: My error, forgot to change this when moved this block from Clusters. Review is really cool thing !

**Comment by dajoRH** [comment]
Lint report for commit aba256847c93450bb403176064ab8fbce5faef3e:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 0291c2b5d7d113292c407145ffc744f7d23b0829:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit e15ccf040ae4f2a10df2e788bdb3600775814090:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit e98c6aa1b013dd6f2170f2ab95355933aaad4acc:

`cfme/infrastructure/cluster.py`:
- :bangbang: Line 25:1: **E112** *expected an indented block*
- :bangbang: Line 25:3: **E227** *missing whitespace around bitwise or shift operator*
- :bangbang: Line 25:3: **E901** *IndentationError: expected an indented block*
- :bangbang: Line 25:7: **E225** *missing whitespace around operator*
- :bangbang: Line 26:5: **E113** *unexpected indentation*
- :bangbang: Line 27:3: **E225** *missing whitespace around operator*
- :bangbang: Line 27:5: **E225** *missing whitespace around operator*
- :bangbang: Line 27:7: **E225** *missing whitespace around operator*
- :bangbang: Line 28:7: **E225** *missing whitespace around operator*
- :bangbang: Line 29:5: **E113** *unexpected indentation*

`cfme/infrastructure/host.py`:
- :bangbang: Line 100:50: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 106:54: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 398:29: **E121** *continuation line under-indented for hanging indent*

Whoa
Whoa :rage4: ! That is a lot of errors. Please go fix them :point_up: 

Commit inspection report:

**No commit flaws detected**.
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 9b265ae2d5222b3eb9bba75996f085103d35ff90:

`cfme/infrastructure/host.py`:
- :bangbang: Line 101:50: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 107:54: **E126** *continuation line over-indented for hanging indent*
- :bangbang: Line 398:29: **E121** *continuation line under-indented for hanging indent*

Please, rectify these issues :smirk: .

Commit inspection report:

**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit e15ccf040ae4f2a10df2e788bdb3600775814090:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit bfd216b854d2a3a9929751c81430f24a799d8f7c:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit d083d039668ad6e906f04ca7905a9897beb0adc9:

Everything seems all right from lint perspective :smile: .

Commit inspection report:
* f637ba61ddceb35781d3bf4e853c231bfcf87211
  * Merge commit detected!!! :bangbang:

**The PR might not be acceptable in this state** :rage4:.
There is a merge commit in the branch which can cause some problems during squashing. Please take **EXTRA care** when squashing!
You have multiple commits in the PR. When you want your PR to be merged, please squash them. Do not squash them when working on the pull request as this allows to see the progress!

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit bfd216b854d2a3a9929751c81430f24a799d8f7c:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit f7da53a0815a66a817cd40df78c43b36043b19f2:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 6e5d1f6f9aa8c94ed6a05b8cae173fd9366d3af5:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Pull request closed without merging**